main.py

```diff
@@ -0,0 +1,313 @@
+              start_ori=base_start_ori,
+              kukaId=kukaId)
+    start_joints = arm.joint_pose
+
+    env = Environment(arm, bottle, is_viz=VISUALIZE)
+    start = np.concatenate(
+        [bottle_start_pos, start_joints])
+    # goal joints are arbitrary and populated later in planner
+    goal = np.concatenate(
+        [bottle_goal_pos, [0] * arm.num_joints])
+    xbounds = [-0.4, -0.9]
+    ybounds = [-0.1, -0.9]
+    dx = dy = dz = 0.1
+    dist_thresh = dx
+    # if  the below isn't true, you're expecting bottle to fall in exactly
+    # the same state bin as the goal
+    assert(dist_thresh <= dx)
+    eps = 5
+    da_rad = 8 * math.pi / 180.0
+
+    with open("filtered_start_goals.obj", "rb") as f:
+        start_goals = pickle.load(f)
+
+    # for i, (startb, goalb, start_joints) in enumerate(start_goals):
+    #     print(startb, goal)
+    #     if np.linalg.norm(np.array(startb) - np.array(goalb)) <= dist_thresh:
+    #         print("%d TOO EASY" % i)
+
+    # exit()
+
+    # change bottle with randomly sampled radius, height, fill, friction
+    # radius = 0.025
+    # height = 0.21
+    # fill_prop = 0.3
+    # fric = env.max_fric
+    # new_bottle = Bottle(start_pos=bottle_start_pos,
+    #                     start_ori=bottle_start_ori,
+    #                     fill_prop=fill_prop,
+    #                     fric=fric)
+    # env.bottle = new_bottle
+
+    # run planner and visualize result
+    num_sims_per_action = 20
+    plan_params_sets = env.gen_random_env_param_set(
+        num=num_sims_per_action)
+    num_exec_tests = 10
+    exec_params_set = env.gen_random_env_param_set(
+        num=num_exec_tests)
+
+    for param in plan_params_sets:
+        print(param)
+
+    load_saved_params = True
+    if load_saved_params:
+        with open("sim_params_set.obj", "rb") as f:
+            exec_plan_params = pickle.load(f)
+            exec_params_set = exec_plan_params["exec_params_set"]
+            plan_params_sets = exec_plan_params["plan_params_sets"]
+    else:
+        with open("sim_params_set.obj", "wb") as f:
+            exec_plan_params = dict(exec_params_set=exec_params_set,
+                                    plan_params_sets=plan_params_sets)
+            pickle.dump(exec_plan_params, f)
+
+    single_planner = NaivePlanner(start, goal, env, xbounds,
+                                  ybounds, dist_thresh, eps, da_rad=da_rad,
+                                  dx=dx, dy=dy, dz=dz, visualize=VISUALIZE, sim_mode=NaivePlanner.SINGLE)
+    avg_planner = NaivePlanner(start, goal, env, xbounds,
+                               ybounds, dist_thresh, eps, da_rad=da_rad,
+                               dx=dx, dy=dy, dz=dz, visualize=VISUALIZE, sim_mode=NaivePlanner.AVG,
+                               num_rand_samples=num_sims_per_action)
+
+    # exec_params_set = plan_params_sets[0]
+    single_planner.sim_params_set = plan_params_sets
+    avg_planner.sim_params_set = plan_params_sets[:10]
```

Comment:
Yes you are completely correct. I wanted the environment to have be responsible for generating environment physics parameters, but I also wanted to give the planner flexibility in choosing which physics parameters to use during planning in case I wanted to do some experiments. In hindsight, I think I should just make the environment solely responsible though and just call some environment function to do this.